garrysmod/lua/includes/extensions/client/panel.lua

```diff
@@ -327,24 +327,30 @@ end
 --[[---------------------------------------------------------
 	Name: SizeToContentsY (Only works on Labels)
 -----------------------------------------------------------]]
-function meta:SizeToContentsY( addval )
+function meta:SizeToContentsY( addVal )
 
 	local w, h = self:GetContentSize()
-	if ( !w || !h ) then return end
+	if ( !w or !h ) then return end
 
-	self:SetTall( h + ( addval or 0 ) )
+	local newSize = h + ( addVal or 0 )
+	if ( newSize == self:GetTall() ) then return end

```

Comment:
I originally added those checks to avoid constant PerformLayouts with certain DLabel panels that have autostretch enabled and thought that I could remove the variable to make the changes a bit simpler, but it seems like this change causes the same issue whereas using a variable for caching does not.

Here's an example that I'm using to test this behavior with `vgui_visualizelayout 1`:

```
    local testFrame = vgui.Create("DFrame")
    testFrame:SetSize(400, 400)
    testFrame:SetDeleteOnClose(true)
    local testLabel = vgui.Create("DLabel", testFrame)
    testLabel:Dock(FILL)
    testLabel:SetText("test")
    testLabel:SetAutoStretchVertical(true)
```

Reply:
Wouldn't the proposed changes just make the function not work if the label's size gets changing by whatever means?

For example:
```lua

local testFrame = vgui.Create("DFrame")
testFrame:SetSize(400, 400)
testFrame:SetDeleteOnClose(true)
testFrame:MakePopup()

local testLabel = vgui.Create("DLabel", testFrame)
testLabel:Dock(FILL)
testLabel:SetText("test")
print( "Size 1: ", testLabel:GetTall())
testLabel:SizeToContentsY()
print( "Size 2: ", testLabel:GetTall())
testLabel:SetTall( 60 )
print( "Size 3: ", testLabel:GetTall())
testLabel:SizeToContentsY()
print( "Size 4: ", testLabel:GetTall()) -- Not sized to contents!

```

Reply:
Alright, I believe I've come up with a solution that should both avoid the broken resizing you've shown and prevent any potential infinite layout loops in https://github.com/Facepunch/garrysmod/pull/2089/commits/1648df2002954f55e0f432f3cf98cdcced9af5d4.